docs/deployment-examples/custom-scripts/index.md

```diff
@@ -4,112 +4,106 @@ description: Custom scripts allows you to script anything you want using PowerSh
-```
+## How Your Scripts are Executed by Octopus
 
-```python Python3
-import os
-createartifact("C:\Windows\System32\drivers\etc\hosts", "{}-hosts.txt".format(os.environ["COMPUTERNAME"]))
-```
+The precise details depend on the context within which your script is running, however Octopus follows this general process:
 
-![Deployment Artifact](images/deployment-artifact.png)
+ 1. Octopus transfers the script to the execution environment along with variables, packages, script modules, and anything else required to run the script. This is done via the Tentacle agent or SSH session into a temporary work directory.
+ 2. The Tentacle agent or SSH session invokes the [open-source Calamari project](https://github.com/OctopusDeploy/Calamari) to bootstrap your script providing access to variables and helper functions. _You can see how your scripts are bootstrapped in the [Calamari source code](https://github.com/OctopusDeploy/Calamari/blob/master/source/Calamari.Shared/Integration/Scripting)._
+ 3. Calamari invokes your script, streaming log messages back to the Octopus Server.
+ 4. Any artifacts published by your script are transferred back to the Octopus Server.
+ 5. The temporary work directory is cleaned up.
 
-### Service Message
+### Working Directories {#Customscripts-Workingdirectories}
 
-The following service message can be written directly (substituting the properties with the relevant values) to standard output which will be parsed by the server and the artifact retrieved at the end of the step. Note that the properties must be supplied as a base64 encoded UTF-8 string.
-```
-##octopus[createArtifact path='<Base64Encoded-FullPath>' name='<Base64Encoded-FileName>' length='<Base64Encoded-FileLength>']
-```
+When Calamari executes your scripts, it does so within the context of a working directory. The working directory is a temporary location stored under the home folder you configure when setting up a deployment target or worker.
+
+If you're executing a script contained within a package, the package contents will be uncompressed and copied to this directory, but the working directory is the directory containing the script within it. After the script has finished executing, the working directory will be cleaned up. Learn about [copying the contents of the working directory](/docs/support/copy-working-directory.md).
 
-## Security and Permissions {#Customscripts-Securityandpermissions}
+### Security and Permissions {#Customscripts-Securityandpermissions}
 
-Keep in mind that scripts are executed in the context of the account that the Tentacle Windows Service (which invokes Calamari) or SSH session runs as.
+Keep in mind that scripts are executed in the context of the account that the Tentacle agent or SSH session runs as. Learn about [running Tentacle as a different user account](/docs/infrastructure/deployment-targets/windows-targets/running-tentacle-under-a-specific-user-account.md).
```

Comment:
I'm having a bit of trouble parsing this first sentence, I know what you mean but `runs as` trips me up, or maybe there too many sticky (connecting) words. I want to say, `...in the context of the account that is running the Tentacle agent or SSH session`, but that might not be quite right.  Maybe:
```suggestion
When scripts are executed, it is in the context of the account that the Tentacle agent or SSH session is running as. Learn about [running Tentacle as a different user account](/docs/infrastructure/deployment-targets/windows-targets/running-tentacle-under-a-specific-user-account.md).
```